server/routes/teacherRegistration.js

```diff
@@ -0,0 +1,51 @@
+import express, { json } from 'express';
+import db from "../db/connection.js";
+import User from "../db/schemas.js";
+
+const router = express.Router();
+router.use(express.json());
+router.use(express.urlencoded({extended: true})); 
+
+
+async function validateId(user_id) {
+    try {
+      const user = await User.findById(user_id);
+      return user ? false : true; 
+    } catch (error) {
+      console.error(error);
+      return false; 
+    }
+  }
+
+function validatePassword(pw){
+    return true;
+}
+
+router.post("/teacher", async(req,res) => {
+    if (req.body != null){
+        const newUser = new User({
+            fname: req.body.fname,
+            lname: req.body.lname,
+            role: "Teacher",
+            user_id: req.body.user_id,
+            pw: req.body.pw
+        });
+        try {
+            // Validate the input
+            if (validatePassword(req.body.password) && validateId(req.body.user_id)) {
```

Comment:
since `validateId` is an async function, I think that you need to add the `await` keyword to make sure we wait for it to be done before we continue with the rest of the code